public/app/plugins/datasource/tempo/QueryField.tsx

```diff
@@ -62,6 +62,12 @@ class TempoQueryFieldComponent extends React.PureComponent<Props, State> {
       serviceMapDatasourceUid: serviceMapDsUid,
       serviceMapDatasource: serviceMapDs as PrometheusDatasource,
     });
+
+    // Set initial query type to ensure traceID field appears
+    this.props.onChange({
+      ...this.props.query,
+      queryType: DEFAULT_QUERY_TYPE,
```

Comment:
Would it make sense to check if queryType already exists in `props.query`? Not sure if it's passed when the URL changes. Also do we still need to set DEFAULT_QUERY_TYPE below as a fallback?

Reply:
From what I've seen, it is not passed in with the URL, but I think it's smart to check if it exists before setting it here. And yeah, in that case I can remove the default fallback below. I'll make those changes.